File: Day 4/Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() -1 , best = 0,width, h, area;
        while (left < right){
            width = right - left;
            if (height[left] < height[right]){
                h = height[left];
            }else{
                h = height[right];
            }
            area = width *h;
            if (area > best){
                best = area;
            }
            if(height[left] < height[right]){
                left +=1;
            }else{
                right -=1;
            }
        }
        return best;
    }
};

File: Day 4/Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        best = 0
        while left < right:
            width = right - left
            h = height[left] if height[left] < height[right] else height[right]

            area = width*h
            if area > best:
                best = area

            if height[left] < height[right]:
                left +=1
            else:
                right -=1
        return best
